### PR TITLE
Make compatible with ghc 8.8 and latest dependencies

### PR DESCRIPTION
**Pull request by locallycompact**
Hi. I wasn't sure what to do about MonadFail but this at least builds.

When this is OK I'll do 8.10 as well.

### CODE REVIEW

File: composite-aeson/src/Composite/Aeson/Formats/DateTime.hs [hunk 1]
```diff
@@ -94,13 +94,14 @@ instance Applicative SuccessOrFail where
...
 instance Monad SuccessOrFail where
   return = Success
-  fail   = Fail
-
   Success a >>= k = k a
   Fail    f >>= _ = Fail f
 
+instance MonadFail SuccessOrFail where
+  fail   = Fail
+
```
Comment (Dridus) [inline composite-aeson/src/Composite/Aeson/Formats/DateTime.hs:104]: could you `#if MIN_VERSION_base(...)` so this builds on older bases as well?

File: composite-aeson/src/Composite/Aeson/Formats/Provided.hs [hunk 2]
```diff
@@ -5,6 +5,7 @@ import Composite.Aeson.Base (JsonFormat(JsonFormat), JsonProfunctor(JsonProfunct
 import Composite.Aeson.Formats.Generic (SumStyle, abeJsonFormat, aesonJsonFormat, jsonArrayFormat, jsonObjectFormat, jsonSumFormat)
 import Composite.Aeson.Formats.InternalTH (makeTupleFormats, makeNamedTupleFormats)
 import Control.Arrow (first)
+import Control.Monad.Except
```
Comment (Dridus) [inline composite-aeson/src/Composite/Aeson/Formats/Provided.hs]: please use an explicit import list (I think it's just `(throwError)`)

File: composite-base/src/Control/Monad/Composite/Context.hs [hunk 3]
```diff
@@ -152,8 +151,6 @@ instance Monad m => Monad (ContextT c m) where
     a <- runContextT clt ctx
     runContextT (k a) ctx
 
-  fail = ContextT . const . fail
-
```
Comment (Dridus) [inline composite-base/src/Control/Monad/Composite/Context.hs:156]: `#if MIN_VERSION_base` guard here too please

File: stack.yaml [hunk 4]
```diff
@@ -9,7 +9,9 @@ packages:
   - example
 flags: {}
 extra-deps:
+  - aeson-1.5.2.0
```
Comment (Dridus) [inline stack.yaml]: why is this extra-depped in rather than using the lts-16.4 version?

**Review by Dridus — CHANGES_REQUESTED** [review]
thanks for the PR!

File: stack.yaml (hunk 4 above)
Comment (locallycompact) [reply stack.yaml]: Whoops this was just me testing. Thanks for the feedback I'll get all these in later.

**Comment by locallycompact** [comment]
This might be correct now.

**Comment by locallycompact** [comment]
Ok now it's working. I added the old manifest as stack-8.6.5.yaml

**Review by Dridus — CHANGES_REQUESTED** [review]
just the explicit import statement, and good to go!

**Comment by locallycompact** [comment]
Ok, good I hope

**Review by Dridus — APPROVED** [review]
lgtm, thank you!

**Comment by locallycompact** [comment]
Could there be a new hackage release with these new bounds?

**Comment by Dridus** [comment]
@locallycompact v0.7.2.0 uploaded to hackage

**Comment by locallycompact** [comment]
Thanks very much. Is it OK if I add these to stackage nightly?

**Comment by Dridus** [comment]
sure if you want to check off the stackage requirements!

**Pull request merged**